app/src/behaviors/behavior_hold_tap.c

```diff
@@ -645,6 +685,11 @@ static int keycode_state_changed_listener(const zmk_event_t *eh) {
         return ZMK_EV_EVENT_BUBBLE;
     }
 
+    if (undecided_hold_tap->first_press && undecided_hold_tap->config->hold_while_undecided) {
```

Comment:
This seems to be the only place where 'first_press' is used and I do not understand why it is needed at all. What does this code solve? Why should we not capture the events that would be captured on line 697?